freedomofpress/securedrop-client
# Move CSS to CSS files and add new tests for stylesheets

# Description

Towards https://github.com/freedomofpress/securedrop-client/issues/1078

Goals behind this PR:
* Start using `.css` files instead of strings or dictionaries of strings or dictionaries of dictionaries of strings
* Separate css stylesheets from Qt application logic
* Use standard CSS object names that correspond to Qt Widget class names
* Set the stylesheet for the application once, except for when we need to change the CSS of an object based on an action.
* Add integration tests to make sure the styles are what we expect
* Add regression tests to make sure if a class name changes then the CSS object name needs to be updated and vice versa

# Test Plan

There are many integration tests that you'll see check to see that the widget styles are what we expect them to be. We are almost at 100% UI style coverage now. Since it's difficult to keep track of where we have coverage for stylesheets, I left comments in the test code for assertions that will get us to 100% coverage. The comments all relate to the following types of tests:

* border tests, including border-radius tests, see https://code.woboq.org/qt5/qtbase/src/widgets/styles/qstylesheetstyle.cpp.html#_ZN21QStyleSheetBorderDataC1EPiP6QBrushPN4QCss11BorderStyleEP5QSize
* padding tests
* text alignment tests
* `menu-indicator` tests
* `image: none;` and `show-decoration-selected: 0;`  tests

There are also comments in the test code for unexpected test failures, e.g. a background color of a disabled button should return gray but instead returns white. There aren't that many of these and it could be because of test writing fatigue that these are broken, however, in order to make sure the client looks like what we expect it to look like, the test plan is to compare master to this pr branch by following this plan:

1. Run the client from `master` and take screenshots
2. Run the client from this PR branch and take screenshots
3. Use a tool to compare the screenshots or manually compare and pay close attention to borders, colors, and layout spacing/ margins.
4. Run the client from qubes and repeat steps 1-3 

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
